bootable-jar/WFLY-14411_Bootable_Jar_Upgrade.adoc

```diff
@@ -0,0 +1,210 @@
+= [Bootable JAR] Server components upgrade and Galleon feature-pack dependency update.
+:author:           Jean-Francois Denise
+:email:             jdenise@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+We have a need to be able to upgrade the server components present in a Bootable JAR when building the jar
+from the Bootable JAR maven plugin.
+
+2 kind of components can be upgraded:
+
+* JBoss Modules module jar files (eg: undertow-core, postgresql driver that would be provisioned from third party Galleon feature-pack).
+* Galleon feature-packs that are dependencies of the Galleon feature-pack(s) in use to build a Bootable JAR.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14411[WFLY-14411]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFGP-194
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:{jkasik@redhat.com}[Jan Kasik]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [X] QE
+
+=== Affected Projects or Components
+
+* Bootable JAR Maven plugin
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [ ] Traditional standalone server (unzipped or provisioned by Galleon)
```

Comment:
This upgrade/update principle is not related to galleon patching. It would be usable for a server provisioned with Galleon Maven plugin but could be not applied to zipped distribution (no provisioning in the picture there).
There is no RFE I am aware of for EAP 8. @bstansberry FYI.

Reply:
So everywhere where galleon provisioning is in place (OpenShift s2i, bootable jar) this would be preferred way in future? 
Where galleon patching will be introduced, legacy patching will be removed?

But we cannot get rid of legacy patching in traditional server distribution. 

Do I understand situation correctly?

Reply:
This feature really targets Bootable JAR. Longer term we could apply it to galleon provisioning by possibly introducing new galleon maven tooling (wildfly specific). For zipped installation, galleon/bootable JAR tooling can't apply, so I would say yes, the current tooling for patching applies.

Reply:
Thank you for context provided here [1]. I will continue in this thread. So this is alternative approach to Galleon patching. So this is deliberately shift away from Galleon patching. Does that mean we shouldn't expect reincarnation of [2] in future? 

My thinking is if we had once Galleon Patching we would want to use it on most possible places. Or this have still its place for bare metal as replacement for legacy patching. And "maven patching" and "galleon patching" will live together? So we will have always 2 ways of patching bare metal and provisioning one.

[1] https://github.com/wildfly/wildfly-proposals/pull/372#issuecomment-801773459
[2] https://issues.redhat.com/browse/EAP7-1063

Reply:
For longer term, @bstansberry can comment. If we want a single patching mechanism to address all, the approach in this proposal is not correct, it is highly bound to Maven and quite specific to Bootable JAR maven plugin.